src/components/vm/disks/diskEdit.tsx

```diff
@@ -20,193 +20,243 @@
 import React from 'react';
 import cockpit from 'cockpit';
 
-import type { Dialogs } from 'dialogs';
-import type { optString, VM, VMDisk, VMDiskDevice } from '../../../types';
+import type { VM, VMDisk } from '../../../types';
 
 import { Button } from "@patternfly/react-core/dist/esm/components/Button";
 import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
-import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
 import {
     Modal, ModalBody, ModalFooter, ModalHeader
 } from '@patternfly/react-core/dist/esm/components/Modal';
-import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
 
-import { ModalError } from 'cockpit-components-inline-notification.jsx';
-import { useDialogs, DialogsContext } from 'dialogs.jsx';
+import { useDialogs } from 'dialogs.jsx';
+import {
+    useDialogState,
+    DialogError, DialogErrorMessage,
+    DialogValue,
+    DialogRadioSelect, DialogRadioSelectOption,
+    DialogDropdownSelectObject,
+    DialogActionButton,
+    DialogCancelButton,
+} from 'cockpit/dialog';
 
 import { domainUpdateDiskAttributes } from '../../../libvirtApi/domain.js';
 import { diskBusTypes, diskCacheModes, getDiskPrettyName, getDiskFullName } from '../../../helpers.js';
 import { NeedsShutdownAlert } from '../../common/needsShutdown.jsx';
-import { InfoPopover } from '../../common/infoPopover.jsx';
 
 const _ = cockpit.gettext;
 
-interface DialogValues {
-    cacheMode: optString;
-    busType: optString;
-    access: string;
-}
-
-type OnValueChanged = <K extends keyof DialogValues>(key: K, value: DialogValues[K]) => void;
-
-const NameRow = ({
+const Name = ({
     idPrefix,
-    name,
-    diskType
+    disk,
 } : {
     idPrefix: string,
-    name: optString,
-    diskType: optString,
+    disk: VMDisk,
 }) => {
     let label = _("ID");
-    if (diskType == "file" || diskType == "block" || diskType == "dir")
+    if (disk.type == "file" || disk.type == "block" || disk.type == "dir")
         label = _("Path");
-    else if (diskType === "network")
+    else if (disk.type === "network")
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2466-9900a690-20251229-144551-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)